worker/buildbot_worker/test/unit/test_runprocess.py

```diff
@@ -857,3 +857,20 @@ def test_statFile_exists(self):
         self.assertEqual(
             st and st[2], 2, "statfile.log exists and size is correct")
         os.remove('statfile.log')
+
+    def test_invalid_utf8(self):
+        # create the log file watcher first
+        rp = self.makeRP()
+        lf = runprocess.LogFileWatcher(rp, 'test', 'invalid_utf8.log',
+                                       follow=False, poll=False)
+        # now write to the log file
+        INVALID_UTF8 = b'\xff'
+        with open('invalid_utf8.log', 'wb') as f:
+            f.write(INVALID_UTF8)
+        # the watcher picks up the changed log
+        lf.poll()
+        # flush she buffer
+        rp._sendBuffers()
+        # cleanup
+        lf.stop()
+        os.remove('invalid_utf8.log')
```

Comment:
I think we need to put this into `finally:`. If test fails then we'd leave the file.

Reply:
The current tests apparently use pytest-trialtemp to run in the _trial_temp subdirectory, which is also listed in .gitignore. The directry is cleared during each test run so leaving stale files in case of error is not a problem, and arguably a debugging feature. 

A NamedTemporaryFile(..., delete=True) can't be reopened on Windows

Reply:
Alright, now I see that this creates more problems than it solves.. I'll try to see if it's possible to address my concern in another way.